Mega Stone refactor

CypherX opened the pull request:
The goal of this was to decouple mega stones from the `PartyPokemon` and turn them into regular items. This also fixes the real evolution issue where mega stones can be lost and allows for the X/Y stones to exist as separate entities rather than being merged into one stone.

Changes:
- Mega stones are now regular items rather than a property on `PartyPokemon`
- Renamed mega stone images to use actual names instead of pokemon IDs
- Pokedex and pokemon list entries support showing both mega stones (really only affects Mewtwo and Charizard)
- Update function to give player any mega stones they have already obtained, even if it was "lost" due to real evolution

There should be no noticeable changes in mega stone functionality for the player. Planning to add mega stones to the item bag in another PR when/if this gets merged.

Aegyo commented on src/modules/requirements/MegaEvolveRequirement.ts:
```diff
@@ -1,16 +1,29 @@
+        } else if (App.game.party.getPokemonByName(this.name).attack < attackRequired) {
+            return `Needs at least ${attackRequired.toLocaleString('en-US')} attack to Mega Evolve.`;
+        } else {
+            return `${this.name} holds no Mega Stone.`;
```
It's probably better to name the mega stone needed here? Something like
```ts
`${this.name} needs the ${MegaStoneType[this.megaStone]} Mega Stone.`
```

Also I think it should either be shown at the same time as the attack requirement, or override it when the player doesn't have the stone yet.

CypherX replied:
Updated

Aegyo commented on src/modules/items/MegaStoneItem.ts:
```diff
@@ -27,17 +27,17 @@ export default class MegaStoneItem extends Item {
     }
     isSoldOut(): boolean {
-        return App.game.party.alreadyCaughtPokemonByName(this.pokemon) && App.game.party.getPokemonByName(this.pokemon).megaStone;
+        return player.itemList[this.name]();
```
```suggestion
        return player.hasMegaStone(this.megaStone);
```

CypherX commented:
Updated to fix the conflicts/issues from #4141 being merged

Jaaslet commented:
Why have we removed the tooltip from the pokedex and pokemon list?

Jaaslet reviewed (changes requested):
`Red_Orb` and `Blue_Orb` shows up in the tooltip like that.

I am going to merge this, as soon as you have fixed the two problems. So you can just clean up the update function now

CypherX commented:
Fixed tooltip and cleaned up the update function.

> Why have we removed the tooltip from the pokedex and pokemon list?

Two reason:

- I couldn't find a decent way to handle it that wasn't messy (particularly the pokedex tooltip for the two pokemon that have two mega stones)
- The evo item modal tooltip has all of that information now.

Jaaslet reviewed (approved):
Looks good. Thanks for the PR
